### ISSUE TITLE
Parser regression, unexpected token '--'

### ISSUE THREAD

**Issue by gkoz** [netvl/xml-rs#105]
After updating to 0.2 parsing [this file](https://github.com/gkoz/gir-files/blob/f32e511f79039cb89d46676e6869be1b5782de21/GLib-2.0.gir) fails with an error:
```
Unexpected token '--' before ' ' in gir-files/GLib-2.0.gir:165:8
```
The problematic piece:
```xml
    <alias name="Time" c:type="GTime">
      <doc xml:space="preserve">Simply a replacement for time_t. It has been deprecated
since it is not equivalent to time_t on 64-bit platforms
with a 64-bit time_t. Unrelated to #GTimer.

Note that #GTime is defined to always be a 32-bit integer,
unlike time_t which may be 64-bit on some systems. Therefore,
#GTime will overflow in the year 2038, and you cannot use the
address of a #GTime variable as argument to the UNIX time()
function.

Instead, do the following:
|[&lt;!-- language="C" --&gt;
time_t ttime;
GTime gtime;

time (&amp;ttime);
gtime = (GTime)ttime;
]|</doc>
      <type name="gint32" c:type="gint32"/>
    </alias>
```

**Comment by netvl** [comment]
Thanks, indeed there is going to be an error if `--` is not followed by `>` anywhere. Don't know why I did that >_< A fix is underway.

**Comment by netvl** [comment]
Fixed in 0.2.1.
